lib/modules/ai_bot/jobs/regular/create_ai_reply.rb

```diff
@@ -6,10 +6,25 @@ class CreateAiReply < ::Jobs::Base
 
     def execute(args)
       return unless bot_user = User.find_by(id: args[:bot_user_id])
-      return unless bot = DiscourseAi::AiBot::Bot.as(bot_user)
       return unless post = Post.includes(:topic).find_by(id: args[:post_id])
 
-      bot.reply_to(post)
+      kwargs = {}
+      kwargs[:user] = post.user
+      if persona_id = post.topic.custom_fields["ai_persona_id"]
+        persona_id = persona_id.to_i
+        kwargs[:persona_id] = persona_id
+      else
+        kwargs[:persona_name] = post.topic.custom_fields["ai_persona"]
+      end
+
+      begin
+        bot = DiscourseAi::AiBot::Bot.as(bot_user, **kwargs)
+        bot.reply_to(post)
```

Comment:
Could just make this a oneliner too, will throw the BOT_NOT_FOUND either way

```suggestion
        DiscourseAi::AiBot::Bot.as(bot_user, **kwargs).reply_to(post)
```

Reply:
mixed on this, a bit of a personal taste here, I don't mind this extra verbosity in this particular case

Reply:
Fair enough!